collect_app/src/main/java/org/odk/collect/android/formmanagement/finalization/EditedFormFinalizationProcessor.kt

```diff
@@ -0,0 +1,37 @@
+package org.odk.collect.android.formmanagement.finalization
+
+import org.javarosa.core.model.FormDef
+import org.javarosa.core.model.data.StringData
+import org.javarosa.core.model.instance.TreeElement
+import org.javarosa.core.util.PropertyUtils
+import org.javarosa.form.api.FormEntryFinalizationProcessor
+import org.javarosa.form.api.FormEntryModel
+import org.odk.collect.android.instancemanagement.isEdit
+import org.odk.collect.forms.instances.Instance
+
+class EditedFormFinalizationProcessor(
+    private val instance: Instance?
+) : FormEntryFinalizationProcessor {
+    override fun processForm(formEntryModel: FormEntryModel) {
+        if (instance?.isEdit() == true) {
+            addDeprecatedId(formEntryModel.form)
+        }
+    }
+
+    private fun addDeprecatedId(formDef: FormDef) {
+        val mainInstance = formDef.mainInstance
+        val metaSection = mainInstance.root.getFirstChild("meta")
+        if (metaSection != null) {
+            val instanceId = metaSection.getFirstChild("instanceID")
+            if (instanceId != null) {
```

Comment:
As far as I can tell there, there's non tests for any of the branching logic here. If I comment out the `instanceId != null` or `deprecatedId == null` checks for instance, no tests fail.

Reply:
Actually, we don't need those checks. 
`instanceId` is required (see https://getodk.github.io/xforms-spec/#metadata) so it can't be null.
`deprecatedId` is always null on form finalization as that's when it gets set.